plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -331,4 +349,30 @@ private static void flushPreferences(IEclipsePreferences preferences) {
       this.shell = shell;
     }
   }
+
+  @VisibleForTesting
+  String jsonEncode(PingEvent event) {
+    Map<String, Object> root = new HashMap<>();
+    Map<String, Object> clientInfo = new HashMap<>();
+    Map<String, String> desktopClientInfo = new HashMap<>();
+    desktopClientInfo.put("os", System.getProperty("os.name"));
```

Comment:
There should be some place to put `AnalyticsPingManager.getPlatformInfo()`?

Reply:
Added it to event.metadata